### PR TITLE
implemented NotImplementedError in xsoar linter

### PR DESCRIPTION
**Pull request by orhovy**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/31061

## Description
Added new xsoar linter validation for "NotImplementedError"

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/lint/resources/pylint_plugins/xsoar_level_checker.py [hunk 1]
```diff
@@ -34,6 +42,31 @@ def _type_annotations_checker(self, node):
...
+            for subnode in list(node.get_children()):
+                yield subnode
+                for sub in self._inner_search_return_error(subnode):
+                    yield sub
+
+        except AttributeError:
+            yield node
+
+        except TypeError:
+            yield node
```
Comment (bakatzir) [inline demisto_sdk/commands/lint/resources/pylint_plugins/xsoar_level_checker.py:68]:
```suggestion
        except (AttributeError, TypeError):
            yield node
```

File: demisto_sdk/commands/lint/resources/pylint_plugins/xsoar_level_checker.py [hunk 2]
```diff
@@ -34,6 +42,31 @@ def _type_annotations_checker(self, node):
...
                 self.add_message("missing-arg-type-annoation", node=node)
 
+    def _not_implemented_error_in_main(self, node):
+        if node.name == 'main':
+            not_implemented_error_exist = False
+            for child in self._inner_search_return_error(node):
+                if isinstance(child, astroid.If):
+                    else_cluse = child.orelse
+                    for line in else_cluse:
+                        if isinstance(line, astroid.Raise) and line.exc.func.name:
```
Comment (bakatzir) [inline demisto_sdk/commands/lint/resources/pylint_plugins/xsoar_level_checker.py:52]: it just checks that an exception is raised, correct?
can we do add an str search for `NotImplementedError`?

File: demisto_sdk/commands/lint/tests/test_pylint_plugin/xsoar_level_checker_test.py [hunk 3]
```diff
@@ -111,3 +111,51 @@ def test_num2(a, b: bool, c:int) -> bool: #@
...
+                ),
+        ):
+            self.checker.visit_functiondef(node)
+
+    def test_not_implemented_error_exists(self):
+        """
+        Given:
+        When:
+        Then:
+        """
```
Comment (bakatzir) [inline demisto_sdk/commands/lint/tests/test_pylint_plugin/xsoar_level_checker_test.py:146]: please tend to this

File: demisto_sdk/commands/lint/resources/pylint_plugins/xsoar_level_checker.py (hunk 2 above)
Comment (orhovy) [reply demisto_sdk/commands/lint/resources/pylint_plugins/xsoar_level_checker.py:52]: yes. That is what I was going for on the right side of the if , accidentally forgot to add it. Thanks for noticing

**Review by bakatzir — COMMENTED** [review]
please add 1 more test case where it does not catch a different exception e.g `DemistoException`/`ValueError`

**Comment by orhovy** [comment]
> please add 1 more test case where it does not catch a different exception e.g `DemistoException`/`ValueError`

I have updated an existing test with more Exceptions

**Review by bakatzir — APPROVED** [review]

**Pull request merged**